backend/src/main/java/com/github/EazyErik/controller/CustomDrinksController.java

```diff
@@ -0,0 +1,38 @@
+package com.github.EazyErik.controller;
+
+
+import com.github.EazyErik.datalayer.CustomDrink;
+import com.github.EazyErik.service.CustomDrinksService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.Principal;
+import java.util.List;
+
+@RestController
+@RequestMapping("api/customDrink")
+@RequiredArgsConstructor
+public class CustomDrinksController {
+
+    private final CustomDrinksService customDrinksService;
+
+
+
+    @PostMapping()
+    public CustomDrink addCustomDrink(@RequestBody CustomDrink customDrink, Principal username) {
+        return customDrinksService.addCustomDrink(customDrink,username.getName());
+
+    }
+    @GetMapping()
+    public List<CustomDrink> getCustomDrinks( @RequestParam(required = false) String ingredient) {
+        return customDrinksService.getCustomDrinks(ingredient);
+    }
+
+    @GetMapping("/details")
+    public CustomDrink getCustomDrink(@RequestParam String id) {
+        return customDrinksService.getCustomDrink(id);
+    }
```

Comment:
Maybe think about using @PathVariable if you only have one Parameter(seems like you only use @RequestParam everywhere). Is't less code and your url is much shorter
```suggestion
    @GetMapping("/details/{id}")
    public CustomDrink getCustomDrink(@PathVariable String id) {
        return customDrinksService.getCustomDrink(id);
    }
```

In your Frontend you can access this endpoint now with `api/customDrink/details/12345`. (replace 12345 with the actual id)